packages/cozy-stack-client/src/__tests__/KonnectorCollection.spec.js

```diff
@@ -0,0 +1,85 @@
+jest.mock('../CozyStackClient')
+
+import CozyStackClient from '../CozyStackClient'
+import KonnectorCollection from '../KonnectorCollection'
+import ALL_KONNECTORS_RESPONSE from './fixtures/konnectors.json'
+
+const FIXTURES = {
+  ALL_KONNECTORS_RESPONSE
+}
+describe(`KonnectorCollection`, () => {
+  const client = new CozyStackClient()
+
+  describe('all', () => {
+    const collection = new KonnectorCollection(client)
+
+    beforeAll(() => {
+      client.fetchJSON.mockReturnValue(
+        Promise.resolve(FIXTURES.ALL_KONNECTORS_RESPONSE)
+      )
+    })
+
+    it('should call the right route', async () => {
+      await collection.all()
+      expect(client.fetchJSON).toHaveBeenCalledWith('GET', '/konnectors/')
+    })
+
+    it('should return a correct JSON API response', async () => {
+      const resp = await collection.all()
+      expect(resp).toConformToJSONAPI()
+    })
+
+    it('should return normalized documents', async () => {
+      const resp = await collection.all()
+      expect(resp.data[0]).toHaveDocumentIdentity()
+    })
+  })
+
+  describe('find', () => {
+    it('throw error', async () => {
+      const collection = new KonnectorCollection(client)
```

Comment:
In don't see the point of these tests. 

You should test : 
- Public interfaces (what a dev should be able to use)
- Expectable dev errors (the dev should not use the interface that way but it may, so we test the returning error)
- Regression or changes proof (the code had another behaviour, we test that the behaviour has changed)

I don't see why we are confirming that calling non existent methods of a new module are returning an exception

Reply:
I've just duplicated tests for AppCollection.

I don't see any input from dev except calling the method `stackClient.konnectors.all()`, but I could be missing a point.

Actually no behavior is changed, we are just adding a namespace for a `KonnectorCollection`. I wanted to highlight this in the last commit by explicitly returning a `DocumentCollection()` for doctype `io.cozy.konnectors` instead of relying on the default behaviour of the switch/case in `CozyStackClient`. By the way I may explicitly test this to ensure no regression is made. I may allow remove the last commit without changing the behaviour but I feel like make a comment here is more future-proof.